app/services/embedded_content_finder_service.rb

```diff
@@ -24,23 +25,44 @@ def find_content_references(value)
 
   def live_content_ids(content_references)
     found_editions = live_editions(content_references.uniq)
-    not_found_content_ids = content_references.map(&:content_id) - found_editions.map(&:content_id)
+    not_found_content_ids = content_references.map(&:identifier) - found_editions.map(&:content_id)
 
     if not_found_content_ids.any?
       GovukError.notify(CommandError.new(
                           code: 422,
                           message: "Could not find any live editions for embedded content IDs: #{not_found_content_ids.join(', ')}",
                         ))
     end
-    content_references.map(&:content_id) - not_found_content_ids
+    content_references.map(&:identifier) - not_found_content_ids
+  end
+
+  def transform_aliases_to_content_ids(content_references)
+    embedded_aliases = content_references.select(&:identifier_is_alias?).map(&:identifier)
+    content_id_aliases = ContentIdAlias.where(name: embedded_aliases).map { |a| [a.name, a.content_id] }.to_h
+    content_references.map do |reference|
+      if reference.identifier_is_alias?
+        identifier = content_id_aliases[reference.identifier]
+        if identifier.nil?
+          GovukError.notify(CommandError.new(
```

Comment:
minor/question: I wonder if we can't find the content_id if we don't return the Reference, as we can't do anything with it without the id? Or we strip them out where `identifier` is nil in `live_content_ids`?

Reply:
Good idea! 👍